### PR TITLE
Fixed #25745 -- Promoted RuntimeWarnings to errors in the test suite.

### PR DESCRIPTION
**Pull request by charettes**

### CODE REVIEW

**Comment by charettes** [comment]
This patch focuses on being non-invasive (for the backport to 1.8) but the schema tests would definitely benefit from a refactoring. I spent way more time than expected trying to figure out which bits to flip.

File: tests/foreign_object/tests.py [hunk 1]
```diff
@@ -417,15 +421,21 @@ class Child(models.Model):
                 related_name='children',
             )
 
+            class Meta:
+                apps = apps = test_apps
```
Comment (timgraham) [inline tests/foreign_object/tests.py]: extra `apps =`

**Comment by charettes** [comment]
@timgraham what should be done in regard to tests that make sure `RuntimeWarning` are raised? Should they be adapted to catch them directly now?

**Comment by timgraham** [comment]
Yes, you can probably make them use `assertRaisesMessage` instead. I don't mind if that commit isn't backported if it proves troublesome.

**Comment by charettes** [comment]
Should be easy to backport if required.

**Comment by charettes** [comment]
Comments addressed.

**Comment by charettes** [comment]
I'll merge tomorrow if you don't beat me to it.

**Pull request merged**